packages/govern-subgraph/schema.graphql

```diff
@@ -63,37 +63,37 @@ type ERC20 @entity {
   address: Bytes!
 }
 
-type QueuePacket @entity {
+type Payload @entity {
   id: ID!
   queue: GovernQueue!
-  status: PacketStatus!
+  status: Status!
```

Comment:
Should we use the term “state” to reflect the contracts?

Also it seems that the hierarchy is the following in the contracts (using TS to describe the relations):

```ts
type Queue = {
  initConfig: Config

  // In the contract only the state is stored in the queue using
  // the container hash as a key, but the subgraph could be more
  // straightforward and have containers in the queue directly.
  queued: Container[]
}

type Config = {
  executionDelay: BigInt
  scheduleDeposit: Collateral
  challengeDeposit: Collateral
  vetoDeposit: Collateral
  resolver: string
  rules: string
}

type Collateral = {
  address: string
  amount: BigInt
}

type Container = {
  config: Config

  // I don’t think it would make sense for a Container to belong to different
  // queues, so I think we can store the state on the Container directly.
  state: ContainerState

  payload: ContainerPayload
}

enum ContainerState {
  None,
  Scheduled,
  Challenged,
  Approved,
  Rejected,
  Cancelled,
  Executed
}

type ContainerPayload = {
  nonce: BigInt
  executionTime: BigInt
  submitter: string
  executor: Executor
  actions: Action[]
  allowFailuresMap: string
  proof: string
}

type Action = {
  to: string
  value: BigInt
  data: string
}
```

Should we try to reproduce this in the subgraph?

Note: I excluded having references to parents for readability, but in the subgraph we could also expose them (e.g. by having `Action.container`).

Reply:
100% agree with this! This is quite similar to the first version of the subgraph, and I think having a structure closer to the contracts lets us understand the subgraph faster.

> Note: I excluded having references to parents for readability, but in the subgraph we could also expose them (e.g. by having Action.container).

Yep, I agree having the parent references in the schema.

Reply:
This structure looks really clean and super close to the smart contract. I diverge a bit from that with the intention to make it a bit more straight forward. And probably has the opposite effect 🙈

I think I miss-understood the importance of having a `Container` that have both a reference to the `Payload` and the `Config`. Because there are scenarios where the `Config` changed and you still want to have the `Payload` linked to the previous `Config`.